pkg/server/options/controllers.go

```diff
@@ -43,18 +43,12 @@ type Controllers struct {
 	SAController kcmoptions.SAControllerOptions
 }
 
-var kcmDefaults *kcmoptions.KubeControllerManagerOptions
-
-func init() {
-	var err error
-
-	kcmDefaults, err = kcmoptions.NewKubeControllerManagerOptions()
+func NewControllers() *Controllers {
+	kcmDefaults, err := kcmoptions.NewKubeControllerManagerOptions()
 	if err != nil {
 		panic(err)
 	}
```

Comment:
I'd prefer to bubble up the error, but that'd require to add error handling to every `NewOptions` that is calling it:

https://github.com/kcp-dev/kcp/blob/90605ce2317922cf3580cab2743c929a76c4541a/pkg/server/options/options.go#L99

https://github.com/kcp-dev/kcp/blob/90605ce2317922cf3580cab2743c929a76c4541a/cmd/kcp/options/options.go#L41

https://github.com/kcp-dev/kcp/blob/90605ce2317922cf3580cab2743c929a76c4541a/cmd/kcp/kcp.go#L90

Which would be a good change - but would also touch a lot of the codebase.

Reply:
I'd rather to that in a separate PR tbh.

Reply:
Panic here is not that bad. It at the point of startup config validation so its quick feedback. Not something happening at runtime 🤷